### PR TITLE
Support upcoming parfive 2.0 release

### PR DESCRIPTION
**Pull request by Cadair**
Currently just testing out the rc with our CI.

### CODE REVIEW

**Comment by nabobalis** [comment]
```
E           parfive.utils.ParfiveFutureWarning: The headers keyword argument to Downloader is deprecated. Please instead pass a SessionConfig object to the config keyword argument.
```

**Comment by Cadair** [comment]
yeah well I should have seen that coming...

**Comment by nabobalis** [comment]
That ignore didnt work for some reason

**Comment by Cadair** [comment]
I think we should merge this PR to main and 4.0, then get a bugfix release of 4.0 out. Then once parfive 2.0 has made it into the wild we can bump the minimum version on main to 2.0 if we want to.

**Review by nabobalis — APPROVED** [review]

File: sunpy/util/parfive_helpers.py [hunk 1]
```diff
@@ -4,27 +4,49 @@
...
+    class Downloader(parfive.Downloader):
+        @wraps(parfive.Downloader.__init__)
+        def __init__(self, *args, **kwargs):
+            if "config" not in kwargs:
+                kwargs["config"] = config
+
+            # This is to make our sample data download code work on 1.x and 2.x
+            # when we depend on 2+ we should remove this.
+            headers = kwargs.pop("headers", {})
+            kwargs["config"].headers = {**headers, **kwargs["config"].headers}
```
Comment (ConorMacBride) [inline sunpy/util/parfive_helpers.py:50]: Are the two dictionaries here the right way around?

Comment (Cadair) [reply sunpy/util/parfive_helpers.py:50]: It should be overriding the `headers` argument with the one in `config`?

Comment (Cadair) [reply sunpy/util/parfive_helpers.py:50]: I think you are right this should be the other way around, not that it matters in practise.

```suggestion
            kwargs["config"].headers = {**kwargs["config"].headers, **headers}
```

**Review by ConorMacBride — APPROVED** [review]

**Pull request merged**